servers/castor/edna.nix

```diff
@@ -0,0 +1,121 @@
+{pkgs, lib, config, ...}:
+  let
+    inherit (builtins) toString typeOf toJSON;
+    inherit (lib) collect concatStringsSep;
+    inherit (pkgs) writeText;
+
+    dns_names = with config.networking;
+      concatStringsSep "," [ "${hostName}.${domain}" "staging.edna.serokell.team" ];
+
+    profile = "/nix/var/nix/profiles/per-user/deploy/edna-docker";
+
+    ports = {
+      backend = {
+        api = 9000;
+      };
+      frontend = {
+        http = 80;
+        https = 443;
+      };
+      prometheus = {
+        node = 9100;
+      };
+    };
+
+    pginit = writeText "init.sql"
+    ''
+      CREATE DATABASE edna;
+    '';
+  in
+  {
+    networking.firewall.allowedTCPPorts = collect (a: (typeOf a) == "int") { inherit (ports) frontend prometheus; };
+
+    virtualisation.docker = {
+      enable = true;
+      logLevel = "warn";
+      storageDriver = "overlay2";
+      volumes = [ "pgdata" ];
+      networks.edna = {};
+    };
+
+    systemd.services.docker-frontend.serviceConfig.RestartSec = "5m";
+
+    virtualisation.oci-containers.containers = let
+      commonOptions = {
+        extraOptions = [
+          # Put all nodes on the same private network
+          "--network=edna"
+
+          # PostgreSQL during initdb
+          "--shm-size=256MB"
+        ];
+      };
+
+    in
+    {
+      frontend = {
+        image = "ghcr.io/serokell/edna-backend";
+        imageFile = "${profile}/ghcr.io-serokell-edna-frontend.tar.gz";
+        dependsOn = [ "backend" ];
+
+        volumes = [
+          "/root/caddy:/.caddy"
+        ];
+
+        ports = with ports.frontend; [
+          "${toString http}:${toString http}"
+          "${toString https}:${toString https}"
+        ];
+
+        environment = {
+          API_HOST = "backend";
+          API_PORT = toString ports.backend.api;
+          DNS_DOMAIN = dns_names;
+        };
+      } // commonOptions;
+
+      backend = let
+        configFile = writeText "config.yaml" (toJSON {
+          api = {
+            listen-addr = "*:${toString ports.backend.api}";
+            serve-docs = false;
+          };
+          db = {
+            host = "postgres";
+            database = "edna";
+            user = "postgres";
+            password = "12345";
```

Comment:
In https://github.com/serokell/edna/pull/8/files it's done a bit differently
![image](https://user-images.githubusercontent.com/2556409/110691624-6d435800-81f6-11eb-9edf-6d0aa3b32ce2.png)

I. e. there we expect a connection string where you can specify any parameters you want.

I think for development we will just use some pre-defined config, so we can implement any of these options (one conn string or separate options for `host`, `database`, etc.). What do you prefer?

Reply:
Oh that's fine, I just put it here as a placeholder. I'll change it to whatever we're actually using once we start using it. I don't mind separate options or conn string.

Upstream postgres container requires setting some environment variables to control db initialization: https://github.com/serokell/gemini-infra/blob/mkaito/edna14-castor/servers/castor/edna.nix#L115

Reply:
Okay, then I think we'll merge it with `conn-string` option and it will have to be updated here. https://github.com/serokell/edna/pull/8 requires just 3 tiny fixes, and if @sashasashasasha151 shows up today, most likely we'll merge it today. Let's update this place after we merge we merge https://github.com/serokell/edna/pull/8 to be sure we won't change it.